### PR TITLE
feat(FEC-9323): advertising configuration

### PR DESCRIPTION
**Pull request by yairans**
### Description of the Changes

* Implement `playAdNow` api 

Depends on https://github.com/kaltura/playkit-js/pull/398

### CheckLists

- [x] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [x] test are passing in local environment 
- [x] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/ima.js [hunk 1]
```diff
@@ -540,6 +537,14 @@ class Ima extends BasePlugin implements IMiddlewareProvider, IAdsControllerProvi
     this.eventManager.listen(this.player, this.player.Event.TIME_UPDATE, () => this._onMediaTimeUpdate());
     this.eventManager.listen(this.player, this.player.Event.SEEKING, () => this._onMediaSeeking());
     this.eventManager.listen(this.player, this.player.Event.SEEKED, () => this._onMediaSeeked());
+    if (!this._playAdByConfig()) {
+      this.eventManager.listen(this.player, this.player.Event.AD_BREAK_START, adEvent => this._stateMachine.adbreakstart(adEvent));
```
Comment (yairans) [inline src/ima.js]: problematic.
`AD_BREAK_START` could be triggered from bumper plugin

File: src/ima.js [hunk 2]
```diff
@@ -540,6 +537,22 @@ class Ima extends BasePlugin implements IMiddlewareProvider, IAdsControllerProvi
...
+        }
+      });
+      this.eventManager.listen(this.player, this.player.Event.AD_BREAK_END, adEvent => {
+        if (this._stateMachine.is(State.PLAYING)) {
+          this._stateMachine.adbreakend(adEvent);
+          this._stateMachine.goto(State.DONE);
+        }
+      });
+      this.eventManager.listen(this.player, this.player.Event.ALL_ADS_COMPLETED, adEvent => {
+        this._stateMachine.adscompleted(adEvent);
```
Comment (yairans) [inline src/ima.js]: problematic.
`ALL_ADS_COMPLETED` could be triggered from bumper plugin

File: src/ima.js [hunk 3]
```diff
@@ -540,6 +537,22 @@ class Ima extends BasePlugin implements IMiddlewareProvider, IAdsControllerProvi
...
+          this._stateMachine.adbreakstart(adEvent);
+        }
+      });
+      this.eventManager.listen(this.player, this.player.Event.AD_BREAK_END, adEvent => {
+        if (this._stateMachine.is(State.PLAYING)) {
+          this._stateMachine.adbreakend(adEvent);
+          this._stateMachine.goto(State.DONE);
+        }
+      });
+      this.eventManager.listen(this.player, this.player.Event.ALL_ADS_COMPLETED, adEvent => {
```
Comment (yairans) [inline src/ima.js]: `ALL_ADS_COMPLETED` could be triggered from bumper plugin

File: src/ima-ads-controller.js [hunk 4]
```diff
@@ -32,13 +32,13 @@ class ImaAdsController implements IAdsPluginController {
...
    * Play an ad on demand.
-   * @param {string} adTagUrl - The ad tag url to play.
+   * @param {Array<Object>} adPod - The ad pod to play.
    * @public
    * @returns {void}
    * @memberof ImaAdsController
    */
-  playAdNow(adTagUrl: string): void {
-    this._context.playAdNow(adTagUrl);
+  playAdNow(adPod: Array<Object>): void {
```
Comment (OrenMe) [inline src/ima-ads-controller.js]: use the `PKAdPod` type from playkit

File: src/ima-state-machine.js [hunk 5]
```diff
@@ -555,19 +561,16 @@ function getAdBreakOptions(adEvent: any): Object {
  */
 function getAdBreakType(adEvent: any): string {
   const ad = adEvent.getAd();
-  const podInfo = ad.getAdPodInfo();
```
Comment (OrenMe) [inline src/ima-state-machine.js]: is there a more "less breaking" way to do this? still using IMA API to get type when possible?

File: src/ima.js [hunk 6]
```diff
@@ -781,7 +824,6 @@ class Ima extends BasePlugin implements IMiddlewareProvider, IAdsControllerProvi
       this._stateMachine.loaded();
     } else {
       this._stateMachine.goto(State.DONE);
-      this.logger.warn('Missing ad tag url: create plugin without requesting ads');
```
Comment (OrenMe) [inline src/ima.js]: why remove?

File: src/ima-state-machine.js (hunk 5 above)
Comment (yairans) [reply src/ima-state-machine.js]: done

File: src/ima-ads-controller.js (hunk 4 above)
Comment (yairans) [reply src/ima-ads-controller.js]: done

File: src/ima.js (hunk 6 above)
Comment (yairans) [reply src/ima.js]: changed to regular log

**Pull request merged**